packages/core/src/StellarHandler.js

```diff
@@ -31,28 +34,26 @@ export default class StellarHandler extends StellarCore {
 
     StellarHandler.isProcessing.add(serviceInbox);
     this.log.info(`@StellarHandler: Starting processing`, { inbox: serviceInbox });
-    this._process(serviceInbox, (job) => {
-      const handler = this.messageHandlers[job.data.headers.queueName];
-      if (!handler) {
-        this.log.error(
-          `No handler for ${job.data.headers.queueName} endpoint registered on this microservice, 
-though it processes the ${serviceInbox} queue`, { inbox: serviceInbox });
-        throw new Error(`No handler for ${job.data.headers.queueName}`);
-      }
-
-      return this.messageHandlers[job.data.headers.queueName](job);
-    });
+    this._process(serviceInbox, ({ data }) =>
+      this
+        ._executeMiddlewares(this.allMiddlewares, data)
+        .then(response => this._enqueue(data.headers.respondTo, response))
+        .catch((e) => {
+          this.log.error(cleanError(e), '@StellarHandler ERROR');
```

Comment:
Can it be migrated to the middleware?

Reply:
no - middlewares are executed in sequence
`process -> mw-1 -> mw-2 -> mw-3 -> fn -> mw-3 -> mw-2 -> mw-1 -> enqueue`

Reply:
this means that process/enqueue calls are outside of middleware